masq_lib/src/test_utils/mock_websockets_server.rs

```diff
@@ -399,4 +467,53 @@ mod tests {
             UiUnmarshalError::fmb(second_expected_response).unwrap()
         );
     }
+
+    #[test]
+    fn broadcast_trigger_can_work_together_with_conversational_messages() {
+        let port = find_free_port();
+        let expected_ui_setup_broadcast = UiSetupBroadcast {
+            running: false,
+            values: vec![UiSetupResponseValue {
+                name: "direction".to_string(),
+                value: "to UI".to_string(),
+                status: Set,
+            }],
+            errors: vec![],
+        };
+        let expected_ui_setup_response = UiSetupResponse {
+            running: true,
+            values: vec![],
+            errors: vec![],
+        };
+        let server = MockWebSocketsServer::new(port)
+            .queue_response(expected_ui_setup_response.clone()
+                .tmb(10),
+            )
+            .queue_response(expected_ui_setup_broadcast.clone().tmb(0))
+            .queue_response(UiChangePasswordResponse {}.tmb(11))
+            .queue_response(UiNewPasswordBroadcast {}.tmb(0));
+        let stop_handle = server.start();
+        let ui_setup_request = UiSetupRequest { values: vec![] };
+        let ui_change_password_request = UiChangePasswordRequest {
+            old_password_opt: None,
+            new_password: "abraka".to_string(),
+        };
+        let broadcast_trigger = UiBroadcastTrigger {
```

Comment:
Perhaps add an implementation of Default to UiBroadcastTrigger so that the `position_to_send_the_signal_opt` field doesn't have to be explicitly specified each time?